recipes/svtjpegxs/all/conanfile.py

```diff
@@ -0,0 +1,81 @@
+import os
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.cmake import cmake_layout, CMakeToolchain, CMake
+from conan.tools.files import copy, get, rmdir, apply_conandata_patches, export_conandata_patches
+from conan.tools.microsoft import is_msvc
+
+required_conan_version = ">=2.0.9"
+
+
+class SvtJpegXsConan(ConanFile):
+    name = "svtjpegxs"
+    description = "A JPEG XS (ISO/IEC 21122) compatible software encoder/decoder library"
+    license = "BSD-2-Clause-Patent"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/OpenVisualCloud/SVT-JPEG-XS"
+    topics = ("jpegxs", "codec", "encoder", "decoder", "image", "video")
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+    implements = ["auto_shared_fpic"]
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.libcxx")
+        self.settings.rm_safe("compiler.cppstd")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def build_requirements(self):
+        self.tool_requires("yasm/1.3.0")
+
+    def validate(self):
+        if self.settings.arch not in ["x86", "x86_64"]:
+            # INFO: The upstream mention about only supporting x86, SSE and AVX
+            # https://github.com/OpenVisualCloud/SVT-JPEG-XS/tree/v0.9.0?tab=readme-ov-file#environment-and-requirements
+            raise ConanInvalidConfiguration(f"{self.ref} does not support {self.settings.arch}. Only x86 and x86_64 are supported.")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+        apply_conandata_patches(self)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.cache_variables["BUILD_APPS"] = False
+        tc.generate()
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
```

Comment:
In the logs, I see cpuinfo being vendorized. Maybe we can try to use CCI's package? If it turns out to be too cumbersome we can look into whether keeping the vendorized does make sense though

Reply:
Nice idea, I'll build locally and post the result.

Reply:
Just checked the upstream code is bit more complicated as it uses a custom build for cpuinfo and it built as module. I would keep the current vendorized version.